activerecord/lib/active_record/core.rb

```diff
@@ -499,30 +499,56 @@ def init_with_attributes(attributes, new_record = false) # :nodoc:
 
     ##
     # :method: clone
-    # Identical to Ruby's clone method.  This is a "shallow" copy.  Be warned that your attributes are not copied.
-    # That means that modifying attributes of the clone will modify the original, since they will both point to the
-    # same attributes hash. If you need a copy of your attributes hash, please use the #dup method.
+    # Identical to Ruby's clone method. This is a "shallow" copy.
     #
     #   user = User.first
-    #   new_user = user.clone
-    #   user.name               # => "Bob"
-    #   new_user.name = "Joe"
-    #   user.name               # => "Joe"
+    #   cloned_user = user.clone
+    #   user.object_id == new_user.object_id # => false
     #
-    #   user.object_id == new_user.object_id            # => false
-    #   user.name.object_id == new_user.name.object_id  # => true
+    # Be warned that your attributes are not copied. That means that modifying
+    # attributes of the clone will modify the original, since they will both
+    # point to the same attributes hash.
     #
-    #   user.name.object_id == user.dup.name.object_id  # => false
+    #   user = User.new(name: "Bob")
+    #   cloned_user = user.clone
+    #   cloned_user.name = "Joe"
+    #   user.name # => "Joe"
+    #
+    # This also applies to associations.
+    #
+    #   user.posts == duped_user.posts # => true
```

Comment:
As above, this should be `cloned_user.posts`

Reply:
Yes.